src/psyclone/tests/psyir/transformations/hoist_trans_test.py

```diff
@@ -0,0 +1,184 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2021, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# ----------------------------------------------------------------------------
+# Author: R. W. Ford, STFC Daresbury Lab
+
+'''This module tests the hoist transformation.
+'''
+
+from __future__ import absolute_import, print_function
+import pytest
+
+from psyclone.psyir.backend.fortran import FortranWriter
+from psyclone.psyir.frontend.fortran import FortranReader
+from psyclone.psyir.nodes import Literal, Loop, Assignment, Reference, \
+    IfBlock, ArrayReference
+from psyclone.psyir.symbols import DataSymbol, REAL_TYPE, INTEGER_TYPE, \
+    BOOLEAN_TYPE, ArrayType
+from psyclone.psyir.transformations import HoistTrans, TransformationError
+from psyclone.tests.utilities import Compile
+
+
+# init
+
+def test_init():
+    '''Test a hoist transformation can be succesfully created.'''
+    hoist_trans = HoistTrans()
+    assert isinstance(hoist_trans, HoistTrans)
+
+
+# apply
+
+def test_apply(tmpdir):
+    '''Test the apply method moves the loop invariant assignment out of
+    the loop and places it immediately before the loop.
+
+    '''
+    code = (
```

Comment:
This should really just be a test of the PSyIR transformation rather than using the frontend and backends. I'll accept the frontend as a shortcut but please could you just check the resulting PSyIR directly rather than use the backend.

Reply:
OK, modified, although it is visually nicer to see the change IMHO. I've (obviously) needed to remove the compilation check.